framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -487,6 +488,11 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
                                     + " should be one of: -Ainfer=jaifs, -Ainfer=stubs");
             }
             wholeProgramInference = new WholeProgramInferenceScenes(this);
+            if (!checker.hasOption("warns")) {
+                // Without -Awarns, the inference output may be incomplete, because javac halts
```

Comment:
An alternative to issuing an error, would be to make it so that `-Ainfer` implies `-Awarns`.

Reply:
Yes, that is a possibility. I think that behavior might surprise users (who hadn't read the manual carefully), so I wanted the `-Awarns` command-line argument to be explicit.